scripts/globals/caskets.lua

```diff
@@ -0,0 +1,1096 @@
+                end
+            end
+            if item == 0 or item == nil then
+                items[i] = 4112 -- default to potion
+            else
+                if math.random() < 0.05 then
+                    items[i] = casketItems[zoneId].regionalItems[math.random(1, #casketItems[zoneId].regionalItems)]
+                else
+                    items[i] = item
+                end
+            end
+        end
+
+        setItems(npc, items[1], items[2], items[3], items[4])
+    ----------------------------------------------
+    -- Evolith drops
+    ----------------------------------------------
+    elseif chestType == "evolith" then
+        local evolith = 2783
+        -- NOTE: Not implimented yet and will be incorperated into items once implimented.
+        -- this is mainly here as a means of testing before implimentation.
+    end
+end
+
+----------------------------------------------------------------------------------
+-- Temp item functions
+----------------------------------------------------------------------------------
+-----------------------------------------------------------------------------------------------------
+-- Desc: Returns an temp items id based on the the local variable i.e. npc:getLocalVariable("TEMP1").
+-----------------------------------------------------------------------------------------------------
+local function getTempDrop(npc, tempNum)
+    local query = string.format("[caskets]TEMP" ..tempNum.. "")
+    local var   = npc:getLocalVar(query)
+
+    if var == nil then
+        return 0
+    else
+        return var
+    end
+end
+
+--------------------------------------------------------------------------------------------------------
+-- Desc: Checks to see if the casket is empty after a player removes an item, if so, despawns the casket
+--------------------------------------------------------------------------------------------------------
+local function checkTempChestIsEmpty(npc)
+    local temp1 = npc:getLocalVar("[caskets]TEMP1")
+    local temp2 = npc:getLocalVar("[caskets]TEMP2")
+    local temp3 = npc:getLocalVar("[caskets]TEMP3")
+
+    if temp1 == 0 and temp2 == 0 and temp3 == 0 then
+        removeChest(npc)
+    end
+end
+
+------------------------------------------------------------------------------------------------
+-- Desc: Gives the player the temp item from a casket based on the selection of the csid
+------------------------------------------------------------------------------------------------
+local function giveTempItem(player, npc, tempNum)
+    local tempQuery   = string.format("[caskets]TEMP" ..tempNum.. "")
+    local tempID      = npc:getLocalVar(tempQuery)
+    local zoneId      = player:getZoneID()
+    local ID          = zones[zoneId]
+    local spawnStatus = npc:getLocalVar("[caskets]SPAWNSTATUS")
+
+    if spawnStatus == casketInfo.spawnStatus.DESPAWNED then
+        player:PrintToPlayer("The chest had already disappeared!", 13)
```

Comment:
if this is what you see on retail I guarantee you that is not what you should be using here. PrintToPlayer is only for debugging or custom messaging. it will be a `messageBasic()` (global, all zones) or `messageSpecial()` (zone specific, which means having a textID to use as an offset for all such messages e.g. `MY_ID_VARIABLE_HERE+3`). If this was only here for debugging purposes that needs to be made more obvious. The same goes for any other `PrintToPlayer()` calls in this PR. And as was previously mentioned, we had  msg.lua global that can be required in to avoid the use of magic numbers: 
```lua
player:PrintToPlayer("The chest had already disappeared!", dsp.msg.channel.NS_SAY)
```
https://github.com/DarkstarProject/darkstar/blob/master/scripts/globals/msg.lua#L13
NS_SAY is the same as `/say` text with the speaker removed, which is value 13. No cs event or NPC will be using that in retail ever. Pretty sure the fact it even exists is a fluke of the client programming.

Reply:
Oh you know what, that actually was left behind before i added text id's and i never removed it after i found that caskets did not use it but sturdy pyxis do, the only reason i left it there was because i was convinced it would be, although in my opinion it should have been put in, I can remove it no problem.

Reply:
Now that I look at the line above it, we shouldn't be able to activate a trigger on a despawned NPC anyway.

Reply:
well if its voodoo or an oversite, i dunno, but it works everytime ive tested it

Reply:
how are you clicking on it if it isn't spawned. Packet injection?

Reply:
just return out with no message I guess

Reply:
thats exactly what i was going to do

Reply:
ok all done, let me know if there's anything else you find

Reply:
na I think we're good, anything else can be edited after the fact when we revisit the file to add more zones's stuffs.